### ISSUE TITLE
Types incorrectly highlighted as values

### ISSUE THREAD

**Issue by sheaf** [JustusAdam/language-haskell#133]
Here are a few constructs that incorrectly highlight types as values:

![types_highlighted_as_values](https://user-images.githubusercontent.com/1297748/80496852-04e6e180-896a-11ea-9969-f24d26813411.png)

It should be a case of including the appropriate type highlighting pattern, e.g. `fun_decl` for the foreign import/export statements.

**Comment by sheaf** [comment]
Some more issues are demonstrated in the `TypeSigs` test case:

![typesig_issues](https://user-images.githubusercontent.com/1297748/80780936-f585b500-8b70-11ea-9984-3157a16335a7.png)

I think there was also a regression for multi-line type signatures. I understand the limitation with lookahead over multiple lines, but at the least the part after the type signature used to be highlighted correctly:

![multiline_typesig](https://user-images.githubusercontent.com/1297748/80780995-249c2680-8b71-11ea-96dc-54fb202f7665.png)

(This is part of the `MultiLineTypeSignatures` test case.)

**Comment by liftM** [comment]
Type signatures with more than one space between the name and double-colon also do not highly properly. For example, this will highlight correctly:

```haskell
data Foo = Foo {
  someField :: Int,
  anotherField :: String
}
```

But this will not:

```haskell
data Foo = Foo {
  someField    :: Int,
  anotherField :: String
}
```

**Comment by sheaf** [comment]
@liftM The problem with record fields has since been addressed, see #118. 

As for the issues raised in the OP, the foreign import/export and data family issues have been resolved by 07c4883 and b39c52a respectively, with the remaining issue surrounding type families near the top of my todo list. After that we should be in a good position to cut a new release.

**Comment by sheaf** [comment]
Fixed remaining issues involving type familes in 7217f36.
